### ISSUE TITLE
Newest develop: can't sync to MPEG frame is still a thing

### ISSUE THREAD

**Issue by dasbts** [elebumm/RedditVideoMakerBot#930]
**Describe the bug**
Still getting the error of "can't sync to MPEG frame" in the newest develop branch version. Tried the proposed solution in this thread but it didn't work. How did I get the error in the newest version? I was getting another error which I tried to fix, that error was that './assets/temp/mp3/0-0.part.mp3' was already in use and thus gave a PermissionError (which looped infinitely and flooded the CMD/terminal), so I altered cleanup.py with the following at row 25: 
```
            try:
                os.remove("./assets/temp/mp3/" + file)
            except:
                pass
```

This gave the following "can't sync to MPEG frame" error: 
```
Traceback (most recent call last):
  File "E:\Python\VideoGenerator\newest\RedditVideoMakerBot\main.py", line 61, in <module>
    run_many(int(getenv("TIMES_TO_RUN")))
  File "E:\Python\VideoGenerator\newest\RedditVideoMakerBot\main.py", line 50, in run_many
    main()
  File "E:\Python\VideoGenerator\newest\RedditVideoMakerBot\main.py", line 38, in main
    length, number_of_comments = save_text_to_mp3(reddit_object)
  File "E:\Python\VideoGenerator\newest\RedditVideoMakerBot\video_creation\voices.py", line 49, in save_text_to_mp3
    return text_to_mp3.run()
  File "E:\Python\VideoGenerator\newest\RedditVideoMakerBot\TTS\engine_wrapper.py", line 67, in run
    self.split_post(comment["comment_body"], idx)
  File "E:\Python\VideoGenerator\newest\RedditVideoMakerBot\TTS\engine_wrapper.py", line 82, in split_post
    self.call_tts(f"{idx}-{idy}.part", text_cut)
  File "E:\Python\VideoGenerator\newest\RedditVideoMakerBot\TTS\engine_wrapper.py", line 94, in call_tts
    self.length += MP3(f"{self.path}/{filename}.mp3").info.length
  File "E:\Python\VideoGenerator\newest\RedditVideoMakerBot\venv\lib\site-packages\mutagen\_file.py", line 48, in __init__
    self.load(*args, **kwargs)
  File "E:\Python\VideoGenerator\newest\RedditVideoMakerBot\venv\lib\site-packages\mutagen\_util.py", line 156, in wrapper
    return func(self, h, *args, **kwargs)
  File "E:\Python\VideoGenerator\newest\RedditVideoMakerBot\venv\lib\site-packages\mutagen\id3\_file.py", line 421, in load
    self.info = self._Info(fileobj, offset)
  File "E:\Python\VideoGenerator\newest\RedditVideoMakerBot\venv\lib\site-packages\mutagen\_util.py", line 185, in wrapper
    return func(*args, **kwargs)
  File "E:\Python\VideoGenerator\newest\RedditVideoMakerBot\venv\lib\site-packages\mutagen\mp3\__init__.py", line 402, in __init__
    raise HeaderNotFoundError("can't sync to MPEG frame")
mutagen.mp3.HeaderNotFoundError: can't sync to MPEG frame
``` 

**To Reproduce**
Alter cleanup.py with the following at row 25: 
```
            try:
                os.remove("./assets/temp/mp3/" + file)
            except:
                pass
```
Then try rendering until the error appears. Unfortunately the CMD was cleaned for me so I don't find the thread again, but I will post an example thread as soon as the issue reappears. 

**Expected behavior**
No error. 

**System (please complete the following information):**
- Python Version: Python 3.9
- OS: Windows 10
- App version / Branch: latest develop

**Comment by dasbts** [comment]
This one for example gives that error. 
https://www.reddit.com/r/musicproduction/comments/vun3m4/share_your_sitting_the_voice_in_the_mix_secrets/

**Comment by dasbts** [comment]
It seems like this issue occurs when the fetching of TTS mp3's fails, it get's stuck at 0-0.part.mp3, thus this error occurs at row 94 in engine_wrapper.py, so I added a try: except: there as the part obviously doesn't have any length, leading to error when trying to read the chunks, kinda back at that FFMPEG error we discussed in the Discord the other day, as it has to have a length. 

So somehow this issue seem to be within the splitting of files, which might be related to the bug that sometimes it jumps to only read part of the comments.

**Comment by William9923** [comment]
Hi, What TTS Provider you are using?

**Comment by Plushhh** [comment]
Im using tiktok and getting this error

 File "C:\Users\user\Desktop\Stuff\YT\reddit bot\RedditVideoMakerBot-master\main.py", line 59, in <module>
    run_many(int(getenv("TIMES_TO_RUN")))
  File "C:\Users\user\Desktop\Stuff\YT\reddit bot\RedditVideoMakerBot-master\main.py", line 49, in run_many
    main()
  File "C:\Users\user\Desktop\Stuff\YT\reddit bot\RedditVideoMakerBot-master\main.py", line 37, in main
    length, number_of_comments = save_text_to_mp3(reddit_object)
  File "C:\Users\user\Desktop\Stuff\YT\reddit bot\RedditVideoMakerBot-master\video_creation\voices.py", line 49, in save_text_to_mp3
    return text_to_mp3.run()
  File "C:\Users\user\Desktop\Stuff\YT\reddit bot\RedditVideoMakerBot-master\TTS\engine_wrapper.py", line 67, in run
    self.split_post(comment["comment_body"], idx)
  File "C:\Users\user\Desktop\Stuff\YT\reddit bot\RedditVideoMakerBot-master\TTS\engine_wrapper.py", line 82, in split_post
    self.call_tts(f"{idx}-{idy}.part", text_cut)
  File "C:\Users\user\Desktop\Stuff\YT\reddit bot\RedditVideoMakerBot-master\TTS\engine_wrapper.py", line 94, in call_tts
    self.length += MP3(f"{self.path}/{filename}.mp3").info.length
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\lib\site-packages\mutagen\_file.py", line 48, in __init__
    self.load(*args, **kwargs)
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\lib\site-packages\mutagen\_util.py", line 156, in wrapper
    return func(self, h, *args, **kwargs)
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\lib\site-packages\mutagen\id3\_file.py", line 421, in load
    self.info = self._Info(fileobj, offset)
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\lib\site-packages\mutagen\_util.py", line 185, in wrapper
    return func(*args, **kwargs)
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\lib\site-packages\mutagen\mp3\__init__.py", line 402, in __init__
    raise HeaderNotFoundError("can't sync to MPEG frame")
mutagen.mp3.HeaderNotFoundError: can't sync to MPEG frame

**Comment by Plushhh** [comment]
Whenever I try to rerun it, it starts at 1 out of 4 again and has the same error which isn't optimal.

**Comment by dasbts** [comment]
TikTok, but I tried with the StreamlabsPolly and GoogleTranslate TTS providers too with errors that there was no input and came to the conclusion that it was ofc. Due to the file size being 0 of the part.mp3 it got stuck on.

Skickades från E-post<https://go.microsoft.com/fwlink/?LinkId=550986> för Windows

Från: William ***@***.***>
Skickat: den 10 juli 2022 07:27
Till: ***@***.***>
Kopia: ***@***.***>; ***@***.***>
Ämne: Re: [elebumm/RedditVideoMakerBot] Newest develop: can't sync to MPEG frame is still a thing (Issue #930)


Hi, What TTS Provider you are using?

**Comment by ElliotPadfield** [comment]
I am experiencing the same bug on latest master. No problems before. Suddenly getting:
` File "/Users/ep/Documents/RedditVideoMakerBot/main.py", line 59, in <module>
    run_many(int(getenv("TIMES_TO_RUN")))
  File "/Users/ep/Documents/RedditVideoMakerBot/main.py", line 49, in run_many
    main()
  File "/Users/ep/Documents/RedditVideoMakerBot/main.py", line 37, in main
    length, number_of_comments = save_text_to_mp3(reddit_object)
  File "/Users/ep/Documents/RedditVideoMakerBot/video_creation/voices.py", line 49, in save_text_to_mp3
    return text_to_mp3.run()
  File "/Users/ep/Documents/RedditVideoMakerBot/TTS/engine_wrapper.py", line 67, in run
    self.split_post(comment["comment_body"], idx)
  File "/Users/ep/Documents/RedditVideoMakerBot/TTS/engine_wrapper.py", line 82, in split_post
    self.call_tts(f"{idx}-{idy}.part", text_cut)
  File "/Users/ep/Documents/GitHub/RedditVideoMakerBot/TTS/engine_wrapper.py", line 94, in call_tts
    self.length += MP3(f"{self.path}/{filename}.mp3").info.length
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/mutagen/_file.py", line 48, in __init__
    self.load(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/mutagen/_util.py", line 156, in wrapper
    return func(self, h, *args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/mutagen/id3/_file.py", line 421, in load
    self.info = self._Info(fileobj, offset)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/mutagen/_util.py", line 185, in wrapper
    return func(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/mutagen/mp3/__init__.py", line 402, in __init__
    raise HeaderNotFoundError("can't sync to MPEG frame")
mutagen.mp3.HeaderNotFoundError: can't sync to MPEG frame`
The last mp3 generated has 0 byte filesize
Using TikTok voice
Python 3.10.5 on MacOS Ventura

**Comment by tonywiththetommy** [comment]
change tiktok max length chars to 270 instead of 300

**Comment by sirreyson** [comment]
> change tiktok max length chars to 270 instead of 300

Changing max length chars does seem to fix the issue, will try several iterations to see if the problem doesn't present itself again

**Comment by JasonLovesDoggo** [comment]
Fixed in 2.3
